### PR TITLE
Use DistributedContextPropagator to propagate Activities

### PR DESCRIPTION
**Pull request by suraciii**
Related issue: #6853

This pr also added baggage propagation.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/7443)

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -43,6 +43,7 @@
   <PropertyGroup>
     <!-- System packages -->
     <SystemDiagnosticsPerformanceCounterVersion>5.0.0</SystemDiagnosticsPerformanceCounterVersion>
+    <SystemDiagnosticsDiagnosticSourceVersion>6.0.0</SystemDiagnosticsDiagnosticSourceVersion>
```
Comment (Ilchert) [inline Directory.Build.props:46]: I think you should upgrade all libs (Ms.Extensions.*, and system packages) to 6.0 version to have consistent transitive references to DiagnosticSource and its dependencies.

Comment (suraciii) [reply Directory.Build.props:46]: A commit(https://github.com/dotnet/orleans/commit/1460dce7901415e4c5f39e4c0f677b46b8c54c64) in another branch already did this, I assume that commit will be cherry-picked into main branch, @ReubenBond what do you think?

Comment (ReubenBond) [reply Directory.Build.props:46]: Yes, we don't need to update it here

**Comment by suraciii** [comment]
@ReubenBond Can this PR be merged now? or is there something I need to change?

**Pull request merged**